connectors/s3-connector/src/main/scala/zio/connect/s3/S3Connector.scala

```diff
@@ -7,13 +7,23 @@ import zio.{Trace, ZIO}
 
 trait S3Connector {
```

Comment:
I think the use of Optional for region is a bit of a pain in practice.
Making the region param non optional everywhere and also setting it's default value to us-east-1 would have better UX. (I think you also mentioned this option when we first discussed)
Example:

val defaultRegion = "us-east-1"
def copyObject(region: String = defaultRegion)(implicit trace: Trace): ZSink[Any, S3Exception, CopyObject, CopyObject, Unit]